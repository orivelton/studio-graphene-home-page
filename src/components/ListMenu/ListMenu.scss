@import '../../assets/scss/helper.scss';

.list-menu {
  display: flex;
  flex-direction: row;
  height: 895px;
  &__type {
    list-style: none;
    width: 268px;
    border-right: 1px solid $very-light-grey;
    padding-top: 200px;
    overflow: hidden;
    &:last-of-type {
      border: 0;
    }

    &:nth-child(1) {
      padding-top: 186px;
    }

    &:nth-child(2) {
      padding-top: 319px;
      text-indent: -10px;
    }

    &:nth-child(3) {
      padding-top: 278px;
      text-indent: -8px;
    }

    &:nth-child(4) {
      padding-top: 223px;
    }
  }
  
  &__item {
    padding: 0 38px;
  }

  &__name {
    font-size: 14px;
    color: $night-rider;
    margin-bottom: 8px;
  }

  &__title {
    font-size: 62px;
    line-height: 0.9;
    margin-bottom: 54px;
  }

  &__description {
    font-family: 'Gill Sans';
    color: $grey;
    font-size: 15px;
  }

  &__price {
    width: max-content;
    padding: 5px 10px 0;
    display: flex;
    justify-content: center;
    border: 2px solid $cinderella;
    margin: 12px 0 45px 0;
  }
}

.parallax {
  transition: margin 0.8s ease-in-out
}
