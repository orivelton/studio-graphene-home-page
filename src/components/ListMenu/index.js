import React from 'react';
import './ListMenu.scss';
import listMenu from '../../mock/listMenu';

const ListMenu = () => {

  const buildItem = ({title, description, price}) => (
    <div className="list-menu__item">
      <p className="list-menu__name">{title}</p>
      <span className="list-menu__description">{description}</span>
      <p className="list-menu__price">â‚¬{price}</p>
    </div>
  );

  return (
    <ul className="list-menu">
      {
        listMenu.map((item) => {
          const { title, menu} = item;
            return (
              <li className="list-menu__type">
                <h3 className="list-menu__title uppercase">{title}</h3>
                { menu.map(item => buildItem(item)) }
              </li>
            )
        })
      }
    </ul>
  );
};

export default ListMenu;
