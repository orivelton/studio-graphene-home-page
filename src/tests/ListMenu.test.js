import React from 'react';
import renderer from 'react-test-renderer';
import ListMenu from '../components/ListMenu';

test('renders correctly ListMenu', () => {
  const tree = renderer.create(<ListMenu />).toJSON();
  expect(tree).toMatchSnapshot();
});
