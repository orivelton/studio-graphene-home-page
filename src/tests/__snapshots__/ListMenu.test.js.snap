// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly ListMenu 1`] = `
<ul
  className="list-menu"
>
  <li
    className="list-menu__type"
  >
    <div
      className="parallax"
    >
      <h3
        className="list-menu__title uppercase"
      >
        STARTERS
      </h3>
      <div
        className="list-menu__item"
      >
        <p
          className="list-menu__name"
        >
          QUINOA CROQUETTAS
        </p>
        <span
          className="list-menu__description"
        >
          Quinoa and cheddar croquettas with aji rocotto & pineapple salsa (v)
        </span>
        <p
          className="list-menu__price"
        >
          €
          4.95
        </p>
      </div>
      <div
        className="list-menu__item"
      >
        <p
          className="list-menu__name"
        >
          CHIFA CHICHARRONES
        </p>
        <span
          className="list-menu__description"
        >
          Quinoa and cheddar croquettas with aji rocotto & pineapple salsa (v)
        </span>
        <p
          className="list-menu__price"
        >
          €
          4.95
        </p>
      </div>
      <div
        className="list-menu__item"
      >
        <p
          className="list-menu__name"
        >
          CALAMARES
        </p>
        <span
          className="list-menu__description"
        >
          Quinoa and cheddar croquettas with aji rocotto & pineapple salsa (v)
        </span>
        <p
          className="list-menu__price"
        >
          €
          4.95
        </p>
      </div>
    </div>
  </li>
  <li
    className="list-menu__type"
  >
    <div
      className="parallax"
    >
      <h3
        className="list-menu__title uppercase"
      >
        MAIN COURSES
      </h3>
      <div
        className="list-menu__item"
      >
        <p
          className="list-menu__name"
        >
          EL CLASICO
        </p>
        <span
          className="list-menu__description"
        >
          Quinoa and cheddar croquettas with aji rocotto & pineapple salsa (v)
        </span>
        <p
          className="list-menu__price"
        >
          €
          4.95
        </p>
      </div>
      <div
        className="list-menu__item"
      >
        <p
          className="list-menu__name"
        >
          TIRADITO CALLAO
        </p>
        <span
          className="list-menu__description"
        >
          Quinoa and cheddar croquettas with aji rocotto & pineapple salsa (v)
        </span>
        <p
          className="list-menu__price"
        >
          €
          4.95
        </p>
      </div>
    </div>
  </li>
  <li
    className="list-menu__type"
  >
    <div
      className="parallax"
    >
      <h3
        className="list-menu__title uppercase"
      >
        SIDES
      </h3>
      <div
        className="list-menu__item"
      >
        <p
          className="list-menu__name"
        >
          SUPER POLLO
        </p>
        <span
          className="list-menu__description"
        >
          Quinoa and cheddar croquettas with aji rocotto & pineapple salsa (v)
        </span>
        <p
          className="list-menu__price"
        >
          €
          4.95
        </p>
      </div>
      <div
        className="list-menu__item"
      >
        <p
          className="list-menu__name"
        >
          PATATAS FRITAS
        </p>
        <span
          className="list-menu__description"
        >
          Quinoa and cheddar croquettas with aji rocotto & pineapple salsa (v)
        </span>
        <p
          className="list-menu__price"
        >
          €
          4.95
        </p>
      </div>
    </div>
  </li>
  <li
    className="list-menu__type"
  >
    <div
      className="parallax"
    >
      <h3
        className="list-menu__title uppercase"
      >
        DESSERTS
      </h3>
      <div
        className="list-menu__item"
      >
        <p
          className="list-menu__name"
        >
          ICECREAM
        </p>
        <span
          className="list-menu__description"
        >
          Quinoa and cheddar croquettas with aji rocotto & pineapple salsa (v)
        </span>
        <p
          className="list-menu__price"
        >
          €
          4.95
        </p>
      </div>
      <div
        className="list-menu__item"
      >
        <p
          className="list-menu__name"
        >
          TIRAMISU
        </p>
        <span
          className="list-menu__description"
        >
          Quinoa and cheddar croquettas with aji rocotto & pineapple salsa (v)
        </span>
        <p
          className="list-menu__price"
        >
          €
          4.95
        </p>
      </div>
      <div
        className="list-menu__item"
      >
        <p
          className="list-menu__name"
        >
          CHOCOLATE BROWNIE
        </p>
        <span
          className="list-menu__description"
        >
          Quinoa and cheddar croquettas with aji rocotto & pineapple salsa (v)
        </span>
        <p
          className="list-menu__price"
        >
          €
          4.95
        </p>
      </div>
    </div>
  </li>
</ul>
`;
